server/repository/migrations/sqlite/2023-04-21T10-00_name_table_add_columns_update_types/up.sql

```diff
@@ -0,0 +1,23 @@
+-- These changes were originally inserted into an earlier migration on develop,
+-- which means they won't run when updating to a "programs" branch if the
+-- earlier migrations had already run.
+
+-- 2021-08-15T10-00_create_name_table
+ALTER TABLE name ADD COLUMN type_new TEXT NOT NULL DEFAULT "FACILITY";
```

Comment:
Now, the column has a default constraint that it doesn't have in develop. Since it's more tricky than expected, I would suggest to just leave `type` as it is (since it's not really required right now and was just an additional clean-up...). Otherwise, I don't see another way than recreating the whole table...

Reply:
It's only because it already has a `NOT NULL` constraint, so something has to be set for a DEFAULT value. However, this default gets immediately overwritten by the current `type` value in the next line:
```
UPDATE name SET type_new =  type;
```

But you can't create a column with NOT NULL without a default if there are existing records, and you can't change constraints after they're created (in SQLite).

Is there a problem leaving it with a default value? One should always be specified when adding records, anyway.

Duplicating the table and transferring the data comes with its own headaches and seems riskier to me.

Reply:
Yes, for this reason I would leave the type column as it is. Not sure how likely it is that we ever add another type variant to the name table... I would defer this problem to the point when we really need it and keep the column with the correct constraint till then.

Reply:
Okay have reverted the type change, it's now as it was (with no change to constraint): 0dcae214f